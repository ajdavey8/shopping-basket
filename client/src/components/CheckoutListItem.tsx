import React from 'react'
import styled from 'styled-components';

import { ItemContainer } from './styled-components';

const InfoWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 450px;
    margin: 0 16px;
`


interface CheckoutListItem {
  id: number
  name: string
  price: number
  image: string
  quantity: number
}

const CheckoutListItem = ({ id, name, price, image, quantity }: CheckoutListItem) => (
  <ItemContainer key={id}>
    <img src={image} alt={`${name} image`} />
    <InfoWrapper>
      <h2>{name}</h2>
      <p>£{price}</p>
      <p>{quantity} item{quantity !== 1 && 's'}</p>
      <p>£{price * quantity}</p>
    </InfoWrapper>
  </ItemContainer>
)

export default CheckoutListItem