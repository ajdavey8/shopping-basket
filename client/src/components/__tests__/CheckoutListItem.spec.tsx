import React from 'react';
import { render } from '@testing-library/react'
import CheckoutListItem from '../CheckoutListItem';

const setup = () => {
  const utils = render(
    <CheckoutListItem
      name='product'
      id={1}
      quantity={1}
      price={2}
      image='imageURL'
    />
  )

  return {
    ...utils
  }
}

describe('<CheckoutListItem>', () => {
  it('renders with the passed props', () => {
    const { asFragment } = setup()
    expect(asFragment).toMatchSnapshot();
  })
})