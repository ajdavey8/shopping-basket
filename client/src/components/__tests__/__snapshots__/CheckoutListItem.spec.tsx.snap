// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutListItem> renders with the passed props 1`] = `[Function]`;
